lighthouse-core/audits/seo/font-size.js

```diff
@@ -76,7 +76,7 @@ function getAttributeMap(attributes = []) {
 
   for (let i = 0; i < attributes.length; i += 2) {
     const name = attributes[i].toLowerCase();
-    const value = attributes[i + 1].trim();
```

Comment:
Let's fix the input type to match our new knowledge of the types here: `Array<string|undefined>=`. What follows is that we should do the same check for `name`.

It's preferable to simply not make an entry in this map if either `name` or `value` is an unexpected `undefined`. In either case, the name-value pair is useless, so we might as well not record it.

A closer look (but not necessary for this PR) would be looking at how the `FontSize` gatherer parses this data: https://github.com/GoogleChrome/lighthouse/blob/826301e7959fdfbcd10a34bffb5f6d581df43a1b/lighthouse-core/gather/gatherers/seo/font-size.js#L249, and at the very least updating the types to reflect that `undefined` is a possibility.

---

```js
/**
 * @param {Array<string|undefined>=} attributes
 * @returns {Map<string, string>}
 */
function getAttributeMap(attributes = []) {
  const map = new Map();

  for (let i = 0; i < attributes.length; i += 2) {
    const name = attributes[i];
    const value = attributes[i + 1];
    if (!name || !value) continue;

    map.set(name.toLowerCase(), value.trim());
  }

  return map;
}
```

Reply:
Makes sense, thanks! I'll make that change.

Reply:
Out of curiosity, what does the trailing `=` on the type `Array<string|undefined>=` mean? I checked the TypeScript docs and tried searching around more generally but couldn't find anything.

Reply:
Actually, I think your solution has a minor behavior change. If `value` is only whitespace characters, the current implementation would leave it out (since it's trimmed before testing its truthiness). The suggested change would mean we would add empty string values to the map in those cases.

The fix I pushed takes this case into account.

Reply:
> Out of curiosity, what does the trailing `=` on the type `Array<string|undefined>=` mean?

It's an optional param, equivalent to

```js
@param {Array<string|undefined>} [attributes]
```

It's not called out explicitly in prose in the typescript docs, but it's in the optional param example, labeled as "Closure syntax":

https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#param-and-returns

Reply:
Ah gotcha, thanks for the link and the clarification!